### ISSUE TITLE
spads_controller get_rating not testable when running on Windows WSL

### ISSUE THREAD

**Issue by jauggy** [beyond-all-reason/teiserver#346]
spads_controller get_rating has this
```
    {rating_value, uncertainty} =
      if host_ip != conn_ip do
        BalanceLib.get_user_rating_value_uncertainty_pair(-1, "Duel")
      else
        BalanceLib.get_user_rating_value_uncertainty_pair(target_id, actual_type)
      end
```
When on Windows WSL, the host_ip and conn_ip will not match and therefore you will always get the Duel rating. I have no idea what that check is for but I think it should be removed. Otherwise it is impossible to locally test anything that interacts with the rating endpoint

**Comment by L-e-x-o-n** [comment]
Rating and balance are public endpoints used by spads to get ratings for players. This check makes it only make it useable by SPADS and not by everyone.

**Comment by L-e-x-o-n** [comment]
You can remove it for local tests.

**Comment by jauggy** [comment]
Can I just add a config to dev.exs e.g. devmode: true and then check for it?
```
if host_ip != conn_ip and !devmode do
```

**Comment by L-e-x-o-n** [comment]
Which IP do local SPADS have?

**Comment by geekingfrog** [comment]
I'm wary of adding checks in the code for dev mode, it tends to lead to more convoluted code. If you want to test something you can modify the code locally. If that becomes a bigger problem, we'll need a better solution.
I had a look at the git history but it's useless to understand why there's this conditional :/

**Comment by jauggy** [comment]
It's there so that in production, outside sources cannot call that endpoint. In production SPADS and teiserver have the same IP.

Modifying that code when developing requires you to know that code even exists and that you're supposed to change it when developing. If I modify it every time I develop, it could easily be committed by mistake too.

**Comment by geekingfrog** [comment]
No, `host_ip` is not teiserver ip it is the ip [given by the host of of the lobby](https://github.com/beyond-all-reason/teiserver/blob/f5db7bb6c30842dbead7a9e68abdd0e67a2d2bfb/lib/teiserver_web/controllers/api/spads_controller.ex#L27-L31). This must match the  [`x-real-ip` header](https://github.com/beyond-all-reason/teiserver/blob/f5db7bb6c30842dbead7a9e68abdd0e67a2d2bfb/lib/teiserver/logging/lib/logging_plug.ex#L89-L93) given when doing this request. There should be a way to force spads to set this header to the "correct" value.

If this is meant as an access control measure, we should return an error like 401/403 not some bogus stuff.

**Comment by jauggy** [comment]
Yeah it's really weird it returns a bogus rating instead of an error.

**Comment by jauggy** [comment]
If you can make it work on WSL that would be preferred. Somehow get this `host_ip != conn_ip` to allow WSL devs.

**Comment by L-e-x-o-n** [comment]
II think @geekingfrog has the best solution.
Instead of returning dual rating by default, return a proper HTTP error.
For testing locally you can either edit the code and disable the check or configure SPADS/WSL somehow to use the correct IP.

**Comment by L-e-x-o-n** [comment]
Alternatively, maybe a site config could be used which would disable the IP check and allow anyone to getratings, it could then be changed locally or even on integration server temporarily if needed, however even with this solution a proper error should still be returned if this option is disabled.

**Comment by jauggy** [comment]
> Alternatively, maybe a site config could be used which would disable the IP check and allow anyone to get ratings, it could then be changed locally or even on integration server temporarily if needed, however even with this solution a proper error should still be returned if this option is disabled.

This would cover all bases.

**Comment by L-e-x-o-n** [comment]
Resolved with https://github.com/beyond-all-reason/teiserver/pull/746
